### PR TITLE
[Closed] New Common sleep configuration, Deep sleep refact, new Sleep function

### PR DESCRIPTION
**Pull request by Sonic0**
| Q               | A
| -------------   | ---
| Bug fix?        | yes
| New feature?    | yes (Personalized sleep event)
| Doc update?     | no 
| BC breaks?      | yes (DeepSleep requires common sleep conf struct)
| Deprecations?   | no
| Relative Issues | #

### CODE REVIEW

File: src/DW1000Ng.cpp [hunk 1]
```diff
@@ -1409,7 +1418,9 @@ namespace DW1000Ng {
 		byte aon_cfg0[LEN_AON_CFG0];
 		memset(aon_cfg0, 0, LEN_AON_CFG0);
 		_readBytes(AON, AON_CFG0_SUB, aon_cfg0, LEN_AON_CFG0);
-		/* SPI and PIN WakeUp events are 1 to default */
+		if ((DW1000NgUtils::getBit(aon_cfg0, 4, WAKE_PIN_BIT) && DW1000NgUtils::getBit(aon_cfg0, 4, WAKE_SPI_BIT)) == 0) {
```
Comment (anacromaniac) [inline src/DW1000Ng.cpp]: I think you meant DW1000NgUtils::getBit(aon_cfg0, 4, WAKE_PIN_BIT) == 0 here

File: examples/StandardRTLSTag_TWR/StandardRTLSTag_TWR.ino [hunk 2]
```diff
@@ -256,7 +260,7 @@ void loop() {
                 }
                 
                 /* Sleep until next blink to save power */
-                DW1000Ng::deepSleep();
```
Comment (anacromaniac) [inline examples/StandardRTLSTag_TWR/StandardRTLSTag_TWR.ino]: Shouldn't this remain deepSleep?

Comment (Sonic0) [reply examples/StandardRTLSTag_TWR/StandardRTLSTag_TWR.ino]: sleep modality depends on the configuration that was applied before

Comment (anacromaniac) [reply examples/StandardRTLSTag_TWR/StandardRTLSTag_TWR.ino]: Oh ok, i read now deepsleep is removed as a method.

**Comment by Sonic0** [comment]
Restarting in a new branch

**Pull request closed without merging**